File: recepteur.py
from pymongo import MongoClient
from paho.mqtt import client as mqtt_client

broker = 'broker.emqx.io' # Broker gratuit ouvert à tout le monde
port = 1883 # Port par défaut
topic = "cesi"
client_id = "paulcalimache"
mongoCon = MongoClient("mongodb://localhost:27017")
db = mongoCon["mqttDatabase"]
coll = db["messages"]

def on_connect(client, userdata, flags, rc):
    if rc == 0:
        print("Connected to MQTT Broker!")
    else:
        print("Failed to connect, return code %d\n", rc)

def on_message(client, userdata, msg):
    msgReceived = msg.payload.decode()
    print("> ", msgReceived)
    msgJSON = {
        "topic": topic,
        "message": msgReceived,
    }
    coll.insert_one(msgJSON)
        
def connect_mqtt() -> mqtt_client:
    client = mqtt_client.Client(client_id)
    client.on_connect = on_connect
    client.connect(broker, port)
    return client


def subscribe(client: mqtt_client):
    client.subscribe(topic)
    client.on_message = on_message

client = connect_mqtt()
subscribe(client)
client.loop_forever()

File: recepteur.js
const mqtt = require('mqtt');
const MongoClient = require('mongodb').MongoClient;

// Connect to the MQTT broker
const client = mqtt.connect('mqtt://broker.emqx.io');

client.on('connect', () => {
  // Once connected, subscribe to the cesi topic
  client.subscribe('cesi', (err) => {
    if (!err) {
      console.log('Successfully subscribed to the cesi topic!');
    }
  });
});

// Set up a callback to handle incoming messages
client.on('message', (topic, message) => {
  // Convert the message to a string
  const messageStr = message.toString();

  // Connect to the MongoDB instance
  MongoClient.connect('mongodb://localhost:27017/', (err, client) => {
    if (err) {
      console.error(err);
      return;
    }

    const db = client.db("mqttDB");

    // Insert the message into the messages collection
    db.collection('messages').insertOne({
      topic: topic,
      message: messageStr,
    }, (err, result) => {
      if (err) {
        console.error(err);
        return;
      }

      console.log('Successfully inserted message : "' + messageStr + '" into MongoDB!');
    });
  });
});